packages/extender/src/registry/registry.js

```diff
@@ -0,0 +1,69 @@
+import * as _ from 'lodash';
+import {composeServices} from './composeServices';
+
+class Registry {
+    constructor() {
+        this.registry = {};
+    }
+
+    addOrReplace(type, key) {
+        const registryKey = type + '-' + key;
+
+        let actions = Array.prototype.slice.call(arguments, 2);
+        let action = composeServices(...actions);
+        action.type = type;
+        action.key = key;
+
+        if (action.targets) {
+            action.targets = action.targets.map(t => {
+                if (typeof t === 'string') {
+                    let spl = t.split(':');
+                    return ({id: spl[0], priority: spl[1] ? spl[1] : 0});
+                }
+
+                return t;
+            });
+        }
+
+        this.registry[registryKey] = action;
+
+        return action;
+    }
+
+    add(type, key) {
+        const registryKey = type + '-' + key;
+        if (this.registry[registryKey]) {
+            throw new Error('Entry already exist for key ' + key);
+        }
+
+        return this.addOrReplace.apply(this, arguments);
+    }
+
+    get(type, key) {
+        return this.registry[type + '-' + key];
+    }
+
+    find(filters) {
+        let result = Object.values(this.registry);
```

Comment:
especialy need some unit test on this method in order to refactor the lodash code.